processing/src/main/java/io/druid/segment/CompressedVSizeIndexedSupplier.java

```diff
@@ -0,0 +1,224 @@
+          order
+      );
+      CompressedVSizeIntsIndexedSupplier valueSupplier = CompressedVSizeIntsIndexedSupplier.fromByteBuffer(
+          buffer,
+          order
+      );
+      return new CompressedVSizeIndexedSupplier(offsetSupplier, valueSupplier);
+    }
+    throw new IAE("Unknown version[%s]", versionFromBuffer);
+  }
+
+  public static CompressedVSizeIndexedSupplier fromIterable(
+      Iterable<IndexedInts> objectsIterable,
+      int maxValue,
+      final ByteOrder byteOrder,
+      CompressedObjectStrategy.CompressionStrategy compression
+  )
+  {
+    Iterator<IndexedInts> objects = objectsIterable.iterator();
+    List<Integer> offsetList = new ArrayList<>();
+    List<Integer> values = new ArrayList<>();
+
+    int offset = 0;
+    while (objects.hasNext()) {
+      IndexedInts next = objects.next();
+      offsetList.add(offset);
+      for (int i = 0; i < next.size(); i++) {
+        values.add(next.get(i));
+      }
+      offset += next.size();
+    }
+    offsetList.add(offset);
+    int offsetMax = offset;
+    CompressedVSizeIntsIndexedSupplier headerSupplier = CompressedVSizeIntsIndexedSupplier.fromList(
+        offsetList,
+        offsetMax,
+        CompressedVSizeIntsIndexedSupplier.maxIntsInBufferForValue(offsetMax),
+        byteOrder,
+        compression
+    );
+    CompressedVSizeIntsIndexedSupplier valuesSupplier = CompressedVSizeIntsIndexedSupplier.fromList(
+        values,
+        maxValue,
+        CompressedVSizeIntsIndexedSupplier.maxIntsInBufferForValue(maxValue),
+        byteOrder,
+        compression
+    );
+    return new CompressedVSizeIndexedSupplier(headerSupplier, valuesSupplier);
+  }
+
+
+      @Override
+  public IndexedMultivalue<IndexedInts> get()
+  {
+    return new CompressedVSizeIndexed(offsetSupplier.get(), valueSupplier.get());
+  }
+
+  public static class CompressedVSizeIndexed implements IndexedMultivalue<IndexedInts>
+  {
+    private final IndexedInts offsets;
+    private final IndexedInts values;
+
+
+    CompressedVSizeIndexed(IndexedInts offsets, IndexedInts values)
+    {
+      this.offsets = offsets;
+      this.values = values;
+    }
+
+    @Override
+    public void close() throws IOException
+    {
+      offsets.close();
+      values.close();
+    }
+
+    @Override
+    public Class<? extends IndexedInts> getClazz()
+    {
+      return IndexedInts.class;
+    }
+
+    @Override
+    public int size()
+    {
+      return offsets.size() -1;
+    }
+
+    @Override
+    public IndexedInts get(int index)
+    {
+      final int offset = offsets.get(index);
+      final int size = offsets.get(index + 1) - offset;
+
+      return new IndexedInts()
+      {
+        @Override
+        public int size()
+        {
+          return size;
+        }
+
+        @Override
+        public int get(int index)
+        {
+          if (index >= size) {
+            throw new IllegalArgumentException(String.format("Index[%s] >= size[%s]", index, size));
```

Comment:
None of the other `get` implementations seem to do bounds checking. This can probably be removed, or at least changed to IndexOutOfBounds exception rather than IAE